### PR TITLE
US-56

### PR DESCRIPTION
**Pull request by amkratz**
US-56 "As a user, I would like to enter the name of a repo and have collaborators automatically populated from Github"

Added new classes BraunfelsService, and BraunfelsRepository.
Updated ProjectDialog to include GUI updates and functionality for adding collaborators.
Includes unit test BraunfelsServiceTest for unit testing BraunfelsService class methods.

Acceptance Testing:
 Run Memoranda, click File and select New Project. Try testing "Add by GitHub Repo" using "ser316asu-2018/braunfels". Save the new project by clicking OK. The project will now be displayed including all Collaborators from the specified repository.

### CODE REVIEW

**Comment by kchoadley** [comment]
JUnit tests are not actually running because the dependency in gradle isn't listed and the test file is not in the correct directory (test/java/test). See pull request 35.
JavaDoc missing in 2 places. Better constructor for BraunfelsUser available. 
Otherwise, looks great! and the JUnit tests do pass.

File: src/main/java/memoranda/braunfels/BraunfelsRepository.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ **/
+public class BraunfelsRepository implements Serializable{
+	private static final long serialVersionUID = 1L;
+	private BraunfelsUser owner;
+	private String name;
+	
+	//constructors
+	public BraunfelsRepository(Repository repo){
+		User user = repo.getOwner();
+		owner = new BraunfelsUser(user.getName(), user.getLogin(), user.getEmail(), user.getUrl());
```
[the remark this answers is not in the document]
Comment (amkratz) [reply src/main/java/memoranda/braunfels/BraunfelsRepository.java]: Fixed! Had an old copy of BraunfelsUser

File: src/main/java/memoranda/braunfels/BraunfelsRepository.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		name = repo.getName();
+	}
+	
+	public BraunfelsRepository(User user, String name){
+		this.owner = new BraunfelsUser(user.getName(), user.getLogin(), user.getEmail(), user.getUrl());
```
[the remark this answers is not in the document]
Comment (amkratz) [reply src/main/java/memoranda/braunfels/BraunfelsRepository.java]: Fixed! Had an old copy of BraunfelsUser

File: src/main/java/memoranda/braunfels/BraunfelsRepository.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * This class creates a BraunfelsUser type object
+ * for use within Memoranda.
+ **/
+public class BraunfelsRepository implements Serializable{
+	private static final long serialVersionUID = 1L;
+	private BraunfelsUser owner;
+	private String name;
+	
+	//constructors
```
[the remark this answers is not in the document]
Comment (amkratz) [reply src/main/java/memoranda/braunfels/BraunfelsRepository.java]: Fixed!

File: src/main/java/memoranda/braunfels/BraunfelsRepository.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	public BraunfelsRepository(Repository repo){
+		User user = repo.getOwner();
+		owner = new BraunfelsUser(user.getName(), user.getLogin(), user.getEmail(), user.getUrl());
+		name = repo.getName();
+	}
+
```
[the remark this answers is not in the document]
Comment (amkratz) [reply src/main/java/memoranda/braunfels/BraunfelsRepository.java]: Fixed!

**Pull request merged**